ducktape/cluster/cluster.py

```diff
@@ -16,14 +16,21 @@
 
 
 class ClusterSlot(object):
-    def __init__(self, parent, account, **kwargs):
-        self.parent = parent
+    def __init__(self, account, **kwargs):
         self.account = account
         for k, v in kwargs.items():
             setattr(self, k, v)
 
-    def free(self):
-        self.parent.free(self)
+    def __eq__(self, other):
+        if other is None:
+            return False
+
+        return self.__dict__ == other.__dict__
+
+    def __hash__(self):
+        # convert self.__dict__ to a hashable type
+        TupleRepresentation = collections.namedtuple('TupleRepresentation', [k for k in self.__dict__.keys()])
```

Comment:
This is going to create the type every time this method is called. If you want `TupleRepresentation`, you should move this out of the method (and probably use `_TupleRepresentation` instead as its private). An alternative implementation would just be `return hash(tuple(x.items()))`